src/cli/repl/meta_cmd.rs

```diff
@@ -649,8 +669,503 @@ impl<F: LurkField + DeserializeOwned> MetaCmd<F> {
+        format: "!(load-data <symbol> <string>)",
+        description: &[],
+        example: &["!(load-data x \"my_file\")"],
+        run: |repl, args| {
+            let (sym, path) = repl.peek2(args)?;
+            if !matches!(sym.tag(), Tag::Expr(ExprTag::Sym)) {
+                bail!(
+                    "Bound variable must be a symbol. Got {}",
+                    sym.fmt_to_string(&repl.store, &repl.state.borrow())
+                )
+            }
+            let path = get_path(repl, &path)?;
+            let LurkData::<F> { z_ptr, z_dag } = load(&path)?;
+            let ptr = z_dag.populate_store(&z_ptr, &repl.store, &mut Default::default())?;
+            let binding = repl.store.cons(sym, ptr);
+            repl.env = repl.store.cons(binding, repl.env);
+            Ok(())
+        },
+    };
+}
+
+impl MetaCmd<F> {
+    const DEFPROTOCOL: MetaCmd<F> = MetaCmd {
+        name: "defprotocol",
+        summary: "Defines a protocol",
+        format: "!(defprotocol <symbol> <vars> <body> options...)",
+        description: &[
+            "The protocol body cannot have any free variable besides the ones",
+            "  declared in the vars list. The body must return a pair such that:",
+            "* The first component is a (proper) list containing the CEK inputs",
+            "  that the proof must accept. This list has 6 elements and follow",
+            "  the order (expr-in env-in cont-in expr-out env-out cont-out).",
+            "  The protocol can deny the proof by returning nil instead of a list.",
+            "* The second component is a 0-arg predicate that will run after the",
+            "  proof verification to further constrain the proof, if needed.",
+            "  If this is not necessary, this component can simply be nil instead.",
+            "",
+            "defprotocol accepts the following options:",
+            "  :rc is the reduction count, defaulting to the REPL's one",
+            "  :lang defines the Lang (ignored for now)",
+            "  :descr is a description of the protocol, defaulting to \"\""
+        ],
+        example: &[
+            "!(defprotocol my-protocol (hash pair)",
+            "  (let ((list6 (lambda (a b c d e f) (cons a (cons b (cons c (cons d (cons e (cons f nil))))))))",
+            "        (mk-open-expr (lambda (hash) (cons 'open (cons hash nil)))))",
+            "    (cons",
+            "      (if (= (+ (car pair) (cdr pair)) 30)",
+            "        (list6 (mk-open-expr hash) nil :outermost pair nil :terminal)",
+            "        nil)",
+            "      (lambda () (> (car pair) 10))))",
+            "  :rc 10",
+            "  :descr \"example protocol\")",
+        ],
+        run: |repl, args| {
+            let (name, rest) = repl.store.car_cdr(args)?;
+            let (vars, rest) = repl.store.car_cdr(&rest)?;
+            let (body, props) = repl.store.car_cdr(&rest)?;
+            if !matches!(name.tag(), Tag::Expr(ExprTag::Sym)) {
```

Comment:
All this code would be more legible and robust if there were reasonable API functions like like `is_sym` or something.